spec/unit/data_providers/hiera_data_provider_spec.rb

```diff
@@ -190,6 +190,90 @@ def compile(environment, code = nil)
       end
     end
   end
+
+  describe 'when using explain' do
+    it 'will report config path (original and resolved), data path (original and resolved), and interpolation (before and after)' do
+      compile('hiera_misc', '$target_scope = "with scope"') do |compiler|
+        lookup_invocation = Puppet::Pops::Lookup::Invocation.new(compiler.topscope, {}, {}, true)
+        value = Puppet::Pops::Lookup.lookup('one::lopts_test::hash', nil, nil, nil, nil, lookup_invocation)
+        expect(lookup_invocation.explainer.to_s).to eq(<<EOS)
+Searching for 'lookup_options'
+  Merge strategy hash
+    Data Binding "hiera"
+      No such key: "lookup_options"
+    Data Provider "Hiera Data Provider, version 4"
+      ConfigurationPath "#{environmentpath}/hiera_misc/hiera.yaml"
+      Data Provider "common"
+        Path "#{environmentpath}/hiera_misc/data/common.yaml"
+          Original path: common
+          Found key: "lookup_options" value: {
+            "one::lopts_test::hash" => {
```

Comment:
Please define "specifics" and how it differs from what we have now. I assume that you want the explanation pruned from everything that doesn't relate to the key? If so, please consider the following:

A _lookup_options_ hash is found by a data provider but the key in question is not included. What does that mean to the explainer? I can see two alternatives:
1. Explain that the hash was found and display all keys in it so that the user can discover that the key in question is misspelled (this is what we currently do).
2. Treat this as a "not found" and prune it from the explanation.

I don't like the pruning for two reasons. a) the user will not discover misspelled keys, and b) we create a special format that will differ from other explanations of a hash merge.

My preference for #3 is to simply add an entry to the explanation stating that the merge options stem from a _lookup_options_ hash with no further info. The option in #2 can be used if the user wants specifics beyond that. Suggested option name for #2 is `--explain-options`. It takes no arguments and can be used in combination with --explain.